indico/modules/events/contributions/forms.py

```diff
@@ -80,3 +98,37 @@ class SubContributionForm(IndicoForm):
     def __init__(self, *args, **kwargs):
         self.event = kwargs.pop('event')
         super(SubContributionForm, self).__init__(*args, **kwargs)
+
+
+class ContributionStartDateForm(IndicoForm):
+    start_dt = IndicoDateTimeField(_('Start date'), [DataRequired(),
+                                                     DateTimeRange(earliest=lambda f, field: f.conf.getStartDate(),
+                                                                   latest=lambda f, field: f.conf.getEndDate())],
+                                   with_clear_button=False)
+
+    def __init__(self, *args, **kwargs):
+        self.contrib = kwargs.pop('contrib')
+        self.conf = self.contrib.event_new.as_legacy
+        self.timezone = self.conf.getTimezone()
+        super(ContributionStartDateForm, self).__init__(*args, **kwargs)
+
+    def validate_start_dt(self, field):
+        if field.data + self.contrib.duration > self.conf.getEndDate():
+            raise ValidationError(_('The selected date is incorrect. Either change the date or the contribution '
```

Comment:
As a user this error would piss me off since it doesn't mention *why* it's incorrect.